pkg/controller/migcluster/validation.go

```diff
@@ -87,38 +87,40 @@ func (r ReconcileMigCluster) validate(cluster *migapi.MigCluster) error {
 func (r ReconcileMigCluster) validateURL(cluster *migapi.MigCluster) error {
 	// Not needed.
 	if cluster.Spec.IsHostCluster {
+		log.V(4).Info("skipping url validation as this is a host cluster",
+			"name", cluster.Name, "namespace", cluster.Namespace)
 		return nil
 	}
 
 	if cluster.Spec.URL == "" {
-		cluster.Status.SetCondition(migapi.Condition{
+		cluster.Status.SetAndLogCondition(log, migapi.Condition{
 			Type:     InvalidURL,
 			Status:   True,
 			Reason:   NotSet,
 			Category: Critical,
-			Message:  MissingURLMessage,
-		})
+			Message:  "`url` is missing for the non-host cluster",
+		}, cluster.Namespace, cluster.Name)
 		return nil
 	}
 	u, err := url.Parse(cluster.Spec.URL)
 	if err != nil {
-		cluster.Status.SetCondition(migapi.Condition{
+		cluster.Status.SetAndLogCondition(log, migapi.Condition{
 			Type:     InvalidURL,
 			Status:   True,
 			Reason:   Malformed,
 			Category: Critical,
-			Message:  MalformedURLMessage,
-		})
+			Message:  "`url` is malformed for the non-host cluster",
```

Comment:
should probably pop out the error here in items. Look at other controllers for example on how errors are popped up

Reply:
Done